File: LeetCode/1. Tow Sum/answer.py
# 1. Two Sum
# https://leetcode.com/problems/two-sum/

class BruteForceSolution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(len(nums)):
                if nums[i] + nums[j] == target and i != j:
                    return [i, j]
        return [-1, -1]


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        comp = {}
        for i in range(len(nums)):
            if target - nums[i] in comp:
                return [comp[target - nums[i]], i]
            comp[nums[i]] = i

        return []

File: LeetCode/1. Tow Sum/answer.ts
// 1. Two Sum
// https://leetcode.com/problems/two-sum/

function twoSum(nums: number[], target: number): number[] {
  const map = {};
  for (let i = 0; i < nums.length; i++) {
    if (target - nums[i] in map) {
      return [i, map[target - nums[i]]];
    }
    map[nums[i]] = i;
  }
  return [];
}

function twoSumBruteForce(nums: number[], target: number): number[] {
  for (let i = 0; i < nums.length; i++) {
    for (let j = 0; j < nums.length; j++) {
      if (nums[i] + nums[j] === target) {
        return [i, j];
      }
    }
  }

  return [];
}
